### ISSUE TITLE
[SUGGESTION] Add layers to editor

### ISSUE THREAD

**Issue by BerryOnGit** [DPS2004/Beatblock-issues#33]
**Is your suggestion related to a problem? Please describe.**
Its too hard to work on a level when vfx related events take up the same space as actual notes.

**Describe the solution you'd like**
A simple layer system that either hides or grays out events or event types on a given layer, making working with events on the same beats and angles less convoluted.

**Describe alternatives you've considered**
A grid of checkboxes to toggle the visibility of a certain type of event, separating vfx and note events into their own vfx / gameplay layers.

**By submitting this, I am confirming that I have read the [README](https://github.com/DPS2004/Beatblock-issues/blob/main/README.md) and am using the latest version of Beatblock.**

**Comment by ItsKirby69** [comment]
yes yes yes. I always wanted to have vfx separated from decos and gameplay. I feel like the solution should have buttons that advance or fall back to a layer or even have a textbox in between the two buttons to specify a certain layer to reach.

Don't want to deal with things like this...
![image](https://github.com/user-attachments/assets/c86f4c6c-3407-4fad-a241-a66ddffd8789)

**Comment by astrob4t** [comment]
> **Describe alternatives you've considered** A grid of checkboxes to toggle the visibility of a certain type of event, separating vfx and note events into their own vfx / gameplay layers.

The alternative considered here would not work quite a few cases, layers would be the best way to implement this because of he often circumstance of a ton of different ease triggers on different areas on the grid and you can't multi-select ones that use certain settings. I know about the repeat setting, but in most of my cases it doesn't work for what I'm trying to do.
 Just wanted to add another piece of to this thread.

**Comment by DPS2004** [comment]
This has been implemented for Early Access
